platform/core/src/utils/dicomLoaderService.js

```diff
@@ -18,14 +18,31 @@ const getImageId = imageObj => {
 
 const findImageIdOnStudies = (studies, displaySetInstanceUID) => {
   const study = studies.find(study => {
-    const displaySet = study.displaySets.some(
+    const foundDisplaySet = study.displaySets.some(
       displaySet => displaySet.displaySetInstanceUID === displaySetInstanceUID
     );
-    return displaySet;
+
+    return foundDisplaySet;
+  });
+
+  const displaySet = study.displaySets.find(displaySet => {
+    return displaySet.displaySetInstanceUID === displaySetInstanceUID;
   });
+
```

Comment:
@SimonBiggs can you help me understand the problem and why these changes could address it? The RT sop class handler uses the first instance in its displaySet object. I'm not sure why you need to get the display set and find them again if it's always the first (previous code).

Reply:
When viewing RT structure files an error was intermittently encountered where the structure file being loaded was being returned as a CT DICOM file. This was because the first item was not the RT structure file. This code gets the UIDs of the structure file being requested and then makes sure to load the DICOM file that matches those UIDs, instead of defaulting to the first (which is intermittently incorrect).